src/porepy/numerics/fv/mpfa.py

```diff
@@ -266,11 +266,60 @@ def discretize(self, sd: pp.Grid, data: dict) -> None:
             loc_bnd: pp.BoundaryCondition = self._bc_for_subgrid(
                 active_bound, sub_sd, l2g_faces, active_grid
             )
+
+            # Eta can either be a scalar or a vector. If a vector valued eta is passed,
+            # it will have a length equal to the number of subfaces in the entire grid.
+            # If the grid is partitioned into subgrids, we need to adjust the length of
+            # eta to match the subfaces of the subgrid.
+            if isinstance(eta, np.ndarray):
+                # First finding how many subfaces each face is divided into
+                sub_sd_subcelltopology = pp.fvutils.SubcellTopology(sub_sd)
+                no_unique_subfaces = len(sub_sd_subcelltopology.subfno_unique)
+                no_faces = sub_sd.cell_faces.shape[0]
+                multiplier = int(no_unique_subfaces / no_faces)
+
+                # Initialize array for the global subface indices
+                global_subgrid_subface_indices = np.zeros(
+                    len(l2g_faces) * multiplier, dtype=int
+                )
+
+                # If all faces are divided into n subfaces, the subfaces will be
+                # numerated as follows: Each face index should be represented n times
+                # and multiplied by n. The 2nd, 3rd to n-th representation of the face
+                # index should have 1, 2, ..., n-1 added to them, respectively. Then you
+                # have the subface indexing.
+
+                # Examples:
```

Comment:
Is this the right place for this documentation?

Reply:
Not sure, but I did not find a better place to put it. Suggestions?